client/my-sites/plan-features/index.jsx

```diff
@@ -266,6 +276,10 @@ class PlanFeatures extends Component {
 				}
 			}
 
+			if ( isInSignupTest && displayJetpackPlans ) {
+				billingTimeFrame = planConstantObj.getSignupBillingTimeFrame();
+			}
```

Comment:
Does this make sense as a different property of `planConstantObj`, or is it just different for the test?

I ask because there is precedent for [passing `abtest` to the getter](https://github.com/Automattic/wp-calypso/blob/332aaf4d8e28a702ee9f3cd09db6b17aab279343/client/my-sites/plan-features/index.jsx#L204) then [conditionally returning different results](https://github.com/Automattic/wp-calypso/blob/332aaf4d8e28a702ee9f3cd09db6b17aab279343/client/lib/plans/constants.js#L268-L286) based on the test.

I believe doing something like that would reduce the size of this PR significantly.